### DISCUSSION TITLE
BLE-sensor to empty mailbox automatically

### DISCUSSION THREAD

**Discussion by steuerzentrale** [PricelessToolkit/MailBoxGuard#31] [Ideas]
I am using the Mailbox-sensor and love the fact that I dont walk to the mailbox just to see that it is empty. As of now it is necessary to "confirm" that the mailbox has been emptied by resetting the counter.

Recently I came up with the following idea and would love to hear your feedback on it.

I think of a BLE-tag which is attached to the keychain. The BLE-tag communicates with a BLE-Lora receiver that is placed in the mailbox. Whenever the BLE-tag is in the range of the BLE-Lora receiver, the counter gets automatically resetted and pushes the information to the LORA-gateway via MQTT. Here I found something which could be used as a starting-point https://www.hackster.io/erictsai/lora-tooth-small-ble-sensors-over-wifi-lora-gateways-0aa109 (Part 5#)

**steps how it would work**
BLE-tag connects with BLE-Lora receiver in mailbox
parcels and/or letters are being taken out of the mailbox
PIR-sensor and/or reed-contacts are not triggering the input.counter
BLE-tag looses connection with BLE-Lora receiver
counter is being reset automatically

what do you think of it?

**Comment by PricelessToolkit** [comment]
Hi, I think it will not work for many people, 
1. For example, I live in an apartment so when I go outside I am passing in proximity with the sensor "max 1m" even if I don't take the mail. The neighbors won't understand if I start avoiding mailboxes 😊
2. Adding complexity to avoid only one click "for many its a one click per week" it's worth the effort and adding cost? I am the laziest man in the world!!, but even for me, it's not to hard just to click the reset button on the phone.

**Comment by steuerzentrale** [comment]
thanks for your quick reply. yeah I get the point with just passing the mailbox. For sure it adds cost and complexity, sometimes its just the challenge to automatise something even if its not worth the cost or effort, just because you can ;) Its not representative, I showed a couple of friends the mailbox-sensor and 3 out of 4 instantly said that it is nice but they would find it even better when it detects if the mailbox has been emptied.

**Reply by PricelessToolkit** [reply]
Yes, I totally agree with you, we need to automate everything possible, but the question is whether it will be reliable enough to not create more problems.

**Comment by kotylo** [comment]
how about another reed switch for opening the actual door of the mailbox, that send different message via LoRa and publish empty MQTT status? Seems like easier to implement and reliable :) but that would require probably a new design to figure out which of the two reed switches triggered.
Heck, if space is not an issue, you can place two mailbox sensors, one for the opening, another for door itself. Overkill, but would work.
